volttron/utils/rmq_setup.py

```diff
@@ -129,6 +133,54 @@ def _set_initial_rabbit_config(instance_name):
     """
     global config_opts
     _load_rmq_config()
+
+    rmq_home = config_opts.get("rmq-home")
+    default_dir = os.path.join(os.path.expanduser("~"), "rabbitmq_server")
+    if rmq_home:
+        rmq_install_dir = os.path.dirname(rmq_home)
+    else:
+        rmq_install_dir = default_dir
+    valid_dir = False
+    while not valid_dir:
+        prompt = 'Rabbitmq install directory:'
+        rmq_install_dir = prompt_response(prompt, default=rmq_install_dir)
+        if rmq_install_dir == default_dir and not os.path.exists(default_dir):
+            os.mkdir(default_dir)
+        valid_dir = os.access(rmq_install_dir, os.W_OK)
+        if not valid_dir:
+            print ("Invalid install directory. Directory should exist and "
+                   "should have write access to user")
+
+    rmq_home = os.path.join(rmq_install_dir, rabbitmq_server)
+    if os.path.exists(rmq_home):
+        print("Given directory already contains {}. "
+              "Skipping rabbitmq server install".format(rabbitmq_server))
+        # if existing server attempt to stop
+        stop_rabbit(rmq_home, quite=True)
```

Comment:
What if the directory exists, but rabbit-server is uninstalled? We can check the status, if it reports not installed, then try to reinstall again.

Reply:
In generic install there is no concept of install and uninstall. only unzip or delete the whole directory.  So some one does something like unzip and removed the contents of the directory instead of the directory itself then all subsequent rabbitmq commands would fail. status command wouldn't work either.  I guess I could check if the rabbitmq-server file exists in rmq_home/sbin but I can't think of what else to do.

Reply:
Ok. I just did a non-sudo rabbitmq installation just now, so I get your point.